### PR TITLE
Add  MambaCache features testcases

### PR DESCRIPTION
**Pull request by YZY00Raiser**
## Motivation
This PR aims to comprehensively enhance the test coverage of the Ascend (NPU) backend for the SGLang framework, with the following key objectives:
1. Test Adaptation of existing features and parameters of SGlang
 Added test coverage for MambaCache features adaptation, verifying the functionality and compatibility of each parameter under normal and edge cases.

## Modifications
1、Add test cases for MambaCache features.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(self.expected_output, response.text)
+        return response.text
+
+    def test_mamba_long_sequence(self):
+        self.process = self._launch_server_with_mamba_params(
+            max_mamba_cache_size=1024
+        )
+        try:
+            time.sleep(5)
```
[the remark this answers is not in the document]
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py]: > medium
> 
> The `popen_launch_server` function already waits for the server to be ready by checking its health endpoint. This `time.sleep(5)` is likely redundant and can make tests flaky. It's better to rely on the health check within `popen_launch_server`.
> 
> This comment also applies to similar `time.sleep(5)` calls on lines 125, 166, 176, 187, 197, 207, and 217.

Deleted.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            other_args.extend(["--max-mamba-cache-size", max_mamba_cache_size])
+        if mamba_ssm_dtye is not None:
+            other_args.extend(["--mamba-ssm-dtype", mamba_ssm_dtye])
+        process = popen_launch_server(
+            self.model,
+            DEFAULT_URL_FOR_TEST,
+            timeout=DEFAULT_TIMEOUT_FOR_SERVER_LAUNCH,
+            other_args=other_args,
+        )
+        return process
```
[the remark this answers is not in the document]
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:83]: > medium
> 
> This method has a typo in the `mamba_ssm_dtye` parameter and constructs the argument list in a way that can lead to duplicated command-line arguments. This is confusing and error-prone.
> 
> I suggest refactoring this method for clarity and correctness. The proposed change uses default arguments and builds the `other_args` list directly, avoiding duplication.
> 
> Note that after applying this suggestion, you will need to update the call sites to use the corrected parameter name `mamba_ssm_dtype`.
> 
> ```python
>     def _launch_server_with_mamba_params(
>         self,
>         max_mamba_cache_size=1024,
>         mamba_ssm_dtype="float32",
>         mamba_full_memory_ratio=0.9,
>         mamba_scheduler_strategy="auto",
>         mamba_track_interval=256,
>     ):
>         other_args = [
>             "--trust-remote-code",
>             "--mem-fraction-static",
>             "0.5",
>             "--attention-backend",
>             "ascend",
>             "--disable-cuda-graph",
>             "--mamba-full-memory-ratio",
>             str(mamba_full_memory_ratio),
>             "--max-mamba-cache-size",
>             str(max_mamba_cache_size),
>             "--mamba-ssm-dtype",
>             mamba_ssm_dtype,
>             "--mamba-scheduler-strategy",
>             mamba_scheduler_strategy,
>             "--mamba-track-interval",
>             str(mamba_track_interval),
>             "--tp-size",
>             "8",
>             "--disable-radix-cache",
>         ]
>         process = popen_launch_server(
>             self.model,
>             DEFAULT_URL_FOR_TEST,
>             timeout=DEFAULT_TIMEOUT_FOR_SERVER_LAUNCH,
>             other_args=other_args,
>         )
>         return process
> ```

The name has been changed.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                "text": prompts,
+                "sampling_params": {
+                    "temperature": 0,
+                    "max_new_tokens": 64,
+                },
+            },
+        )
+        results = response.json()
+        for i, result in enumerate(results):
+            self.assertGreater(len(result["text"]), 0)
```
[the remark this answers is not in the document]
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py]: > high
> 
> This test method launches a server process but never kills it, which will lead to a resource leak. All other tests in this file use a `try...finally` block to ensure the process is killed. The same pattern should be applied here.
> 
> ```python
>     def test_mamba_batch(self):
>         # test use mamba in batch requests can work properly
>         self.process = self._launch_server_with_mamba_params(
>             max_mamba_cache_size=512,
>         )
>         try:
>             prompts = [
>                 "What is AI",
>                 "Explain neural network",
>                 "How does deep learning differ from machine learning",
>                 "What is reinforcement learning",
>                 "Explain natural language processing",
>                 "What are neural network layers",
>                 "How do activation functions work",
>                 "Explain backpropagation",
>                 "What is computer vision",
>                 "How do LLMs work",
>             ]
>             response = requests.post(
>                 f"{DEFAULT_URL_FOR_TEST}/generate",
>                 json={
>                     "text": prompts,
>                     "sampling_params": {
>                         "temperature": 0,
>                         "max_new_tokens": 64,
>                     },
>                 },
>             )
>             results = response.json()
>             for i, result in enumerate(results):
>                 self.assertGreater(len(result["text"]), 0)
>         finally:
>             kill_process_tree(self.process.pid)
> ```

The cleanup method has been added.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            other_args.extend(["--mamba-ssm-dtype", mamba_ssm_dtye])
+        process = popen_launch_server(
+            self.model,
+            DEFAULT_URL_FOR_TEST,
+            timeout=DEFAULT_TIMEOUT_FOR_SERVER_LAUNCH,
+            other_args=other_args,
+        )
+        return process
+
+    def _tes_basic_inference(self):
```
[the remark this answers is not in the document]
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py]: > high
> 
> There is a typo in the method name. It should be `_test_basic_inference`. Please also update all call sites of this method (e.g., on lines 126, 167, 177, etc.).

The name has been changed.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_radix_cache.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class TestMambaCache(CustomTestCase):
+    """Testcase：Verify the test kv cache reuse, when use mamba cache.
+
+    [Test Category] Parameter
+    [Test Target] --disable-radix-cache
+    """
+
+    model = QWEN3_NEXT_80B_A3B_INSTRUCT_WEIGHTS_FOR_TEST
```
[the remark this answers is not in the document]
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_radix_cache.py]: > critical
> 
> `QWEN3_NEXT_80B_A3B_INSTRUCT_WEIGHTS_FOR_TEST` is a `ModelTestConfig` object, not a string path. The `popen_launch_server` function expects a string for the model path. You should use `.model_path` to get the actual path.

model_path has been updated.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+            "0.5",
+            "--attention-backend",
+            "ascend",
+            "--disable-cuda-graph",
+            "--mamba-full-memory-ratio",
+            mamba_full_memory_ratio,
+            "--max-mamba-cache-size",
+            "1024",
+            "--mamba-ssm-dtype",
+            "float32",
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:64]: delete R63 R64

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+    def test_mamba_scheduler_no_buffer(self):
+        self.process = self._launch_server_with_mamba_params(
+            mamba_scheduler_strategy="no_buffer",
+        )
+        try:
+            self._test_basic_inference()
+        finally:
+            kill_process_tree(self.process.pid)
+
+    def test_mamba_ssm_dtype_float32(self):
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:170]: Delete redundant test points.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_radix_cache.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
+import unittest
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/MambaCache/test_npu_mamba_radix_cache.py:1]: Add a test case combined with the HiCache feature.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py [hunk 9]
```diff
@@ -0,0 +1,298 @@
+import tempfile
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:1]: Refactor test cases to reduce service startup times through parameter integration.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_radix_cache.py (hunk 8 above)
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_radix_cache.py:1]: > Add a test case combined with the HiCache feature.

Added

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py (hunk 9 above)
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:1]: > Refactor test cases to reduce service startup times through parameter integration.
Integrated already.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py (hunk 7 above)
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:170]: > Delete redundant test points.

Deleted.

File: test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py (hunk 6 above)
Comment (YZY00Raiser) [reply test/registered/ascend/basic_function/MambaCache/test_npu_mamba_cache_combination_parameters.py:64]: > delete R63 R64

Deleted.
